aot/compile.jl

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+# -*- mode: julia -*-
+#=
+exec "${JULIA:-julia}" "$@" ${BASH_SOURCE[0]}
+=#
+
+using Pkg
+Pkg.activate(@__DIR__)
+Pkg.add("MacroTools")
+Pkg.develop(PackageSpec(path=dirname(@__DIR__)))
+Pkg.build("PyCall")
+Pkg.activate()
+
+using PackageCompiler
+sysout, _curr_syso = compile_incremental(
+    joinpath(@__DIR__, "Project.toml"),
+    joinpath(@__DIR__, "precompile.jl"),
+)
+
+pysysout = joinpath(@__DIR__, basename(sysout))
+cp(sysout, pysysout, force=true)
+
+write(joinpath(@__DIR__, "_julia_path"), Base.julia_cmd().exec[1])
+
+@info "System image: $pysysout"
```

Comment:
This all seems like boilerplate — why does PyCall.jl need a custom `compile.jl` file?

Reply:
Snoop in PackageCompiler does not work due to https://github.com/JuliaLang/julia/issues/31156 (a bug in Julia?).  There is another "latent" issue https://github.com/JuliaLang/PackageCompiler.jl/issues/192.  I think PackageCompiler is still rather a moving target so using lower-level API with a bit of boilerplate makes sense.  Another upside is that we can control exactly what functions are included in system image by tweaking code in `aot/precompile.jl` (although it is rather sparse and random ATM).